File: MADVCamera/CameraManager/ambarequest/AMBAPutFileRequest.java
package com.madv360.madv.connection.ambarequest;

import com.google.gson.annotations.Expose;
import com.madv360.madv.connection.AMBARequest;
import com.madv360.madv.connection.ambaresponse.AMBAPutFileResponse;

/**
 * Created by wang yandong on 16/9/20.
 */
public class AMBAPutFileRequest extends AMBARequest {
    public AMBAPutFileRequest(ResponseListener responseListener) {
        super(responseListener, AMBAPutFileResponse.class);
    }

    @Expose
    public long offset;

    @Expose
    public long size;

    @Expose
    public String md5sum;

    @Expose
    public int fType;
}

File: MADVCamera/CameraManager/ambarequest/AMBAPutFileRequest.m
//
//  AMBAPutFileRequest.m
//  Madv360_v1
//
//  Created by QiuDong on 16/11/17.
//  Copyright © 2016年 Cyllenge. All rights reserved.
//

#import "AMBAPutFileRequest.h"
#import "AMBAPutFileResponse.h"

@implementation AMBAPutFileRequest

- (instancetype) initWithReceiveBlock:(AMBAResponseReceivedBlock)receiveBlock errorBlock:(AMBAResponseErrorBlock)errorBlock {
    if (self = [super initWithReceiveBlock:receiveBlock errorBlock:errorBlock responseClass:AMBAPutFileResponse.class])
    {
        
    }
    return self;
}

+ (NSArray<NSString* >*) jsonSerializablePropertyNames {
    mergeJsonSerializablePropertyNames(array, @[@"offset", @"size", @"md5sum", @"fType"]);
    return array;
}

@end
